core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -2602,6 +2611,32 @@ private RelBuilder aggregate_(GroupKeyImpl groupKey,
     return project(projects.transform((i, name) -> aliasMaybe(field(i), name)));
   }
 
+  public static Set<Integer> getAllFields2(ImmutableBitSet groupSet,
+      List<AggregateCall> aggCallList, boolean checkSubquery, int expectedSelectListCount) {
+    final Set<Integer> allFields = new TreeSet<>();
+    allFields.addAll(groupSet.asList());
+    if (checkSubquery) {
+      //In case of pivot with subquery we need all columns of subquery and in-clause fields both
+      //hence we are adding all the fields to allFields.
+      for (int i = 0; i < expectedSelectListCount; i++) {
+        allFields.add(i);
+      }
```

Comment:
instead of for loop iteration like this make it more concise:

```
IntStream.range(0, expectedSelectListCount).forEach(allFields::add);

```

IntStream.range(0, expectedSelectListCount): This generates a stream of integers from 0 (inclusive) to expectedSelectListCount (exclusive).
forEach(allFields::add): This adds each integer to allFields directly.

Reply:
resolved